filebeat/beater/filebeat.go

```diff
@@ -55,10 +55,14 @@ func New(b *beat.Beat, rawConfig *common.Config) (beat.Beater, error) {
 	// Add prospectors created by the modules
 	config.Prospectors = append(config.Prospectors, moduleProspectors...)
 
-	if len(config.Prospectors) == 0 {
+	if !config.ProspectorReload.Enabled() && len(config.Prospectors) == 0 {
 		return nil, errors.New("No prospectors defined. What files do you want me to watch?")
 	}
 
+	if *once && config.ProspectorReload.Enabled() {
+		return nil, errors.New("Once and prospector reloading cannot be used together.")
```

Comment:
Consider changing `s/Once/-once/` and/or swapping it around like "prospector reloading and -once cannot be used together".